app/screens/user_profile/user_info.tsx

```diff
@@ -17,36 +19,63 @@ type Props = {
     showNickname: boolean;
     showPosition: boolean;
     user: UserModel;
+    enableCustomAttributes?: boolean;
 }
 
-const UserInfo = ({localTime, showCustomStatus, showLocalTime, showNickname, showPosition, user}: Props) => {
-    const {formatMessage} = useIntl();
+const emptyList = [] as DisplayCustomAttribute[]; /** avoid re-renders **/
+
+const UserInfo = ({localTime, showCustomStatus, showLocalTime, showNickname, showPosition, user, enableCustomAttributes}: Props) => {
     const customStatus = getUserCustomStatus(user);
+    const serverUrl = useServerUrl();
+    const [customAttributes, setCustomAttributes] = useState<DisplayCustomAttribute[]>([]);
+    const lastRequest = useRef(0);
+
+    useEffect(() => {
+        if (enableCustomAttributes) {
+            const fetchData = async () => {
+                const reqTime = Date.now();
+                lastRequest.current = reqTime;
+                try {
+                    const client = NetworkManager.getClient(serverUrl);
+                    const [fields, attrValues] = await Promise.all([
+                        client.getCustomProfileAttributeFields(),
+                        client.getCustomProfileAttributeValues(user.id),
+                    ]);
+
+                    // ignore results if there was a newer request
+                    if (fields && lastRequest.current === reqTime) {
```

Comment:
One of the implications of this if, if you don't look deeper may be "if there are no files, do nothing". That is not exactly that, because that only happens if fields is undefined. If fields is the empty list, this still updates customAttributes correctly to the empty list.

But if fields is undefined, we do nothing. Is that intended? (Not sure when fields might be undefined)